examples/samd/interrupt/main.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2020, Erik Henriksson
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <modm/board.hpp>
+#include <modm/platform/extint/extint.hpp>
+
+using namespace Board;
+using namespace std::chrono_literals;
+
+static volatile bool blink = true;
+
+void
+isr() {
+  blink = !blink;
+  // Kids, please don't do serial logging in ISRs...
+  MODM_LOG_DEBUG << "blink: " << (blink ? "true" : "false") << modm::endl;
+}
+
+int
+main()
+{
+    Board::initialize();
+    ExternalInterrupt::initialize();
+    ExtInt<3>::initialize(&isr);
+    ExtInt<3>::connectPin<D12>();
```

Comment:
Why not make the interrupt a signal?
```suggestion
    ExternalInterrupt::connect<D12::ExtInt3>(&isr);
```

Reply:
I guess you could even make it a variadic template so that you can attach multiple handlers to the same pin:
```suggestion
    ExternalInterrupt::connect<D12::ExtInt3>(&isr1, &isr2, &isr3);
```
or multiple pins to the same handler?
```suggestion
    ExternalInterrupt::connect<D10::ExtInt1, D11::ExtInt2, D12::ExtInt3>(&isr);
```
It's all just syntactic sugar though…